### PR TITLE
S3 Zone Settings Resource

### PR DESCRIPTION
**Pull request by Krishnan-Priyanshu**
# Description
Resource for s3 zone setting 

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### RESOURCE OR DATASOURCE NAME

powerscale_s3_zone_settings"

image

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Unit tests
- [x] Acceptance tests

### CODE REVIEW

File: docs/resources/s3_zone_settings.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+# Available actions: Create, Update, Delete and Import
+# After `terraform apply` of this example file it will modify S3 Global Settings on  the PowerScale Array.
```
Comment (Krunal-Thakkar) [inline docs/resources/s3_zone_settings.md:51]: s3 zone setting

File: examples/resources/powerscale_s3_zone_settings/resource.tf [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+# Available actions: Create, Update, Delete and Import
+# After `terraform apply` of this example file it will modify S3 Global Settings on  the PowerScale Array.
+# For more information, Please check the terraform state file.
+
+# PowerScale S3 Zone settings allows you to configure S3 Zone settings on PowerScale.
+resource "powerscale_s3_zone_settings" "s3_zone_setting" {
+  zone                         = "tfacc-test"
+  root_path                    = "/ifs/"
+  base_domain                  = ""
```
Comment (Krunal-Thakkar) [inline examples/resources/powerscale_s3_zone_settings/resource.tf:26]: any sample value for base_domain?

File: powerscale/helper/s3_zone_settings_helper.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
+package helper
```
Comment (Krunal-Thakkar) [inline powerscale/helper/s3_zone_settings_helper.go:1]: add copyright

**Review by shenda1 — APPROVED** [review]

**Comment by Krishnan-Priyanshu** [comment]
run e2e test TestAccS3Zone

**Comment by Krishnan-Priyanshu** [comment]
run e2e test S3Zone

**Comment by Krishnan-Priyanshu** [comment]
run e2e test S3Zone

**Review by Krunal-Thakkar — APPROVED** [review]

**Pull request merged**